src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -200,8 +201,34 @@ public final void onCompleted(final Run run, @Nonnull final TaskListener listene
           // The client is a threadpool so instead of creating a new instance of the pool
           // we lease the exiting one registerd with Jenkins.
           statsd = getDescriptor().leaseClient();
-          statsd.incrementCounter("completed", tagsToCounter);
-          logger.fine(String.format("Attempted to send 'completed' counter with tags: %s", Arrays.toString(tagsToCounter)));
+          statsd.incrementCounter("completed", statsdTags);
+          logger.fine(String.format("Attempted to send 'completed' counter with tags: %s", Arrays.toString(statsdTags)));
+
+          logger.fine("Computing KPI metrics");
+          // Send KPIs
+          if (run.getResult() == Result.SUCCESS) {
+            long mttr = getMeanTimeToRecovery(run);
+            long cycleTime = getCycleTime(run);;
+            long leadTime = run.getDuration() + mttr;
+
+            statsd.gauge("leadtime", leadTime / 1000, statsdTags);
+            if (cycleTime > 0) {
+              statsd.gauge("cycletime", cycleTime / 1000, statsdTags);
+            }
+            if (mttr > 0) {
```

Comment:
Same as above and below

Reply:
Its because the functions that calculate these values use 0 as a "nothing to report" kind of special value. Like if there is no previous run that was successful, this metric doesn't make sense, and the function that calculates this value would return a value that doesn't really mean anything. Seems like the same for below.